### PR TITLE
Change rendering tests to detect semantic on the root DesignDoc 

### PR DESCRIPTION
**Pull request by timothyfroehlich**
Targeting main with this PR instead of #459 which targeted the feature branch.

This provides a more consistent method of detecting and testing DesignCompose docs and whether they rendered.

### CODE REVIEW

**Comment by rylin8** [comment]
Confused -- doesn't the screenshot diff show a regression?

**Comment by timothyfroehlich** [comment]
https://github.com/takahirom/roborazzi-compare-on-github-comment-sample/issues/6

**Comment by rylin8** [comment]
Do you have a plan on merging this into the rustlayout branch?

**Comment by timothyfroehlich** [comment]
> Do you have a plan on merging this into the rustlayout branch?

I'm just going to bring this in. The scope of this PR ended up getting narrowed. Originally this included a workaround for the rustlayout branch dropping the modifiers that were passed in, but like you saw yesterday, that caused layout regressions. So I've dropped that workaround and instead this PR is only for changing the semantics and how the tests detect the DesignDoc and whether it's rendering properly. The net result of this is just a better testing API, essentially.


Separately I'm going to finish up my change to get the libraries building for your host and hooked up to the tests, which will allow the screenshot tests to run (and fail). Roborazzi should be able to generate screenshots at that point, but if not then I'll push another change to disable the assertions. That way the test won't fail and Roborazzi should be able to grab the screenshots no problem.

File: designcompose/src/main/java/com/android/designcompose/DesignView.kt [hunk 1]
```diff
@@ -954,7 +951,7 @@ internal fun DesignDocInternal(
             LaunchedEffect(docId) { designComposeCallbacks?.docReadyCallback?.invoke(docId) }
             CompositionLocalProvider(LocalDesignIsRootContext provides DesignIsRoot(false)) {
                 DesignView(
-                    modifier.semantics { sDocId = docId },
+                    modifier.semantics { sDocRenderStatus = docRenderStatus },
```
Comment (rylin8) [inline designcompose/src/main/java/com/android/designcompose/DesignView.kt:954]: How will this part work with in the rustlayout branch?

Comment (timothyfroehlich) [reply designcompose/src/main/java/com/android/designcompose/DesignView.kt:954]: I don't know, and so I'm basically punting that. 

Basically, we don't have a solution for getting the test assertions to work yet. But I'm pretty sure I can get screenshot tests working either way.

I still think this is a good change. The changes that I made to the actual tests will persist to the rustlayout branch. Basically we'll need to figure out how to get the semantics applied in rustlayout, but the tests themselves won't need to change.

Comment (rylin8) [reply designcompose/src/main/java/com/android/designcompose/DesignView.kt:954]: Ok that's fine, you'll just have to deal with some conflicts merging to the feature branch or merging the feature branch back to main.

Comment (timothyfroehlich) [reply designcompose/src/main/java/com/android/designcompose/DesignView.kt:954]: Yep

**Review by rylin8 — APPROVED** [review]

**Pull request merged**